instrumentation/opentelemetry-api-1.0/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/trace/ApplicationSpan.java

```diff
@@ -140,11 +141,20 @@ public boolean isRecording() {
   }
 
   @Override
-  public boolean equals(Object other) {
-    if (!(other instanceof ApplicationSpan)) {
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (!(obj instanceof ApplicationSpan)) {
       return false;
     }
-    return agentSpan.equals(((ApplicationSpan) other).agentSpan);
+    ApplicationSpan other = (ApplicationSpan) obj;
+    return Objects.equals(agentSpan, other.agentSpan);
```

Comment:
I would usually not use Objects for non-nullable fields. Otherwise the non-nullness gets a bit obfuscated. What do you think?

Reply:
yeah, this makes sense, I'll update to use direct `equals` where that was the previous behavior at least